provider/common.go

```diff
@@ -85,31 +88,41 @@ func buildBatchRequest(transactions interface{}, privateKey solana.PrivateKey, o
 	batchRequest := pb.PostSubmitBatchRequest{}
 	batchRequest.SubmitStrategy = opts.SubmitStrategy
 
-	for _, tx := range transactions.([]interface{}) {
-
-		oneRequest := pb.PostSubmitRequestEntry{}
-		oneRequest.SkipPreFlight = opts.SkipPreFlight
-
-		if txStr, ok := tx.(string); ok {
-			signedTxBase64, err := transaction.SignTxWithPrivateKey(txStr, privateKey)
+	switch transactions.(type) {
+	case []*api.TransactionMessage:
+		for _, txMsg := range transactions.([]*api.TransactionMessage) {
+			oneRequest, err := createBatchRequestEntry(opts, txMsg.Content, privateKey)
 			if err != nil {
-				return &pb.PostSubmitBatchRequest{}, err
-			}
-			oneRequest.Transaction = &pb.TransactionMessage{
-				Content: signedTxBase64,
+				return nil, err
 			}
-		} else if txMsg, ok := tx.(*pb.TransactionMessage); ok {
-			signedTxBase64, err := transaction.SignTxWithPrivateKey(txMsg.Content, privateKey)
+
+			batchRequest.Entries = append(batchRequest.Entries, oneRequest)
+		}
+	case []string:
+		for _, txStr := range transactions.([]string) {
+			oneRequest, err := createBatchRequestEntry(opts, txStr, privateKey)
 			if err != nil {
-				return &pb.PostSubmitBatchRequest{}, err
-			}
-			oneRequest.Transaction = &pb.TransactionMessage{
-				Content:   signedTxBase64,
-				IsCleanup: txMsg.IsCleanup,
+				return nil, err
 			}
-		}
 
-		batchRequest.Entries = append(batchRequest.Entries, &oneRequest)
+			batchRequest.Entries = append(batchRequest.Entries, oneRequest)
+		}
+	default:
+		log.Fatalf("transactions has unknown type : %v", reflect.TypeOf(transactions))
```

Comment:
Can we return an error here instead of logging it?

Reply:
best not enable this kind of behavior at all I think –– https://bloxroute.atlassian.net/browse/TRAD-285 for removing this completely